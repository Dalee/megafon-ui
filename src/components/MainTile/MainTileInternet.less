@import '~styles/base.less';

@b: e('.main-tile-internet');

@{b} {
    &__title {
        margin: 0;
    }
    &__text {
        margin-bottom: 4px;
    }
}

