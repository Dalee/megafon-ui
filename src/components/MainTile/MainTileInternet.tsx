import * as React from 'react';
import * as PropTypes from 'prop-types';
import './MainTileInternet.less';
import Header from '../Header/Header';
import DropdownSocialList from '../DropdownSocialList/DropdownSocialList';
import { cnCreate } from '../../utils/cn';
import { ISocialIcon, IShowcaseChildren } from './MainTileFeatures';

interface IMainTileInternetProps {
    /** Custom class name */
    className?: string;
    /** Social icons list */
    socialIcons?: Array<Partial<ISocialIcon>>;
    /** Showcase params childrens list */
    params?: Array<Partial<IShowcaseChildren>>;
}

const cn = cnCreate('main-tile-internet');
class MainTileInternet extends React.Component<IMainTileInternetProps, {}> {
    static propTypes = {
        className: PropTypes.string,
        params: PropTypes.array,
        socialIcons: PropTypes.array,
    };

    static defaultProps: IMainTileInternetProps = {
        params: [],
        socialIcons: [],
    };

    render() {
        const { params, socialIcons }: IMainTileInternetProps = this.props;

        return (
            <div className={cn('', {}, this.props.className)} >
                {params!.map((param: IShowcaseChildren, index: number): React.ReactNode =>
                    <div className={cn('inner')} key={param.title + index}>
                        <Header
                            className={cn('title')}
                            as="h3"
                            margin={false}
                        >
                            {param.title}
                        </Header>
                        <div className={cn('text')}>
                            {param.caption}
                        </div>
                        {!!socialIcons!.length && <DropdownSocialList icons={socialIcons!} />}
                    </div>
                )}
            </div>
        );
    }
}

export default MainTileInternet;
