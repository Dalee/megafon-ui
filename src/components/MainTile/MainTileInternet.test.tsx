import * as React from 'react';
import { shallow } from 'enzyme';
import MainTileInternet from './MainTileInternet';
import Checked from 'icons/checked_24.svg';

const props = {
    title: 'test',
    params: [
        {
            svgIcon: <Checked />,
            title: 'test',
            caption: 'test',
            value: 'test',
        },
    ],
    socialIcons: [{
        svgIcon: <Checked />,
        title: 'test',
    }],
};

describe('<MainTileInternet />', () => {
    it('it renders MainTileInternet', () => {
        const wrapper = shallow(<MainTileInternet {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
