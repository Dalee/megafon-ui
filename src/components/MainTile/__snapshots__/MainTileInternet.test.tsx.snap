// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainTileInternet /> it renders MainTileInternet 1`] = `
<div
  className="main-tile-internet"
>
  <div
    className="main-tile-internet__inner"
    key="test0"
  >
    <Header
      as="h3"
      className="main-tile-internet__title"
      color="black"
      margin={false}
    >
      test
    </Header>
    <div
      className="main-tile-internet__text"
    >
      test
    </div>
    <DropdownSocialList
      icons={
        Array [
          Object {
            "svgIcon": <test-file-stub />,
            "title": "test",
          },
        ]
      }
      maxIconNumber={6}
    />
  </div>
</div>
`;
